vault-plugins/flant_iam_auth/path_jwt_type_test.go

```diff
@@ -0,0 +1,333 @@
+	name := randomStr()
+	resp := jwtTypePathRequester(t).Create(name, params)
+	return name, resp
+}
+
+func mustCreateUpdateJWTType(t *testing.T, body map[string]interface{}) string {
+	name, resp := createJWTType(t, body)
+	code := resp.StatusCode
+	if code != 200 {
+		b, _ := io.ReadAll(resp.Body)
+		t.Errorf("incorrect response code after creating: %v: %v", code, string(b))
+	}
+
+	return name
+}
+
+func getJWTType(t *testing.T, name string) *api.Response {
+	return jwtTypePathRequester(t).Get(name)
+}
+
+func cleanAllJWTTypes(t *testing.T) {
+	keys, _ := jwtTypePathRequester(t).ListKeys()
+
+	for _, n := range keys {
+		jwtTypePathRequester(t).Delete(n)
+	}
+}
+
+func assertJwtType(t *testing.T, resp *api.Response, data map[string]interface{}) {
+	respData := extractResponseData(t, resp)
+
+	if uuid, ok := respData["uuid"]; !ok || uuid == "" {
+		t.Errorf("incorrect uuid")
+	}
+
+	for k, v := range data {
+		rv, ok := respData[k]
+
+		if !ok {
+			t.Errorf("has not key '%s' in response", k)
+		}
+
+		assert.DeepEqual(t, rv, v)
+	}
+}
+
+func TestJWTTypePath(t *testing.T) {
+	skipNoneDev(t)
+
+	jWTTypeCreate(t)
+	jWTTypeGet(t)
+	jWTTypeDelete(t)
+	jWTTypeUpdate(t)
+	jWTTypeList(t)
+}
+
+func jWTTypeCreate(t *testing.T) {
+	t.Run("creating successful", func(t *testing.T) {
+		cases := []struct {
+			title string
+			body  map[string]interface{}
+		}{
+			{
+				title: "creates with all supported fields",
+				body: map[string]interface{}{
+					"ttl":            "1s",
+					"options_schema": testJwtTypeOptionSchemaValid,
+				},
+			},
+
+			{
+				title: "creates without 'options_schema'",
+				body: map[string]interface{}{
+					"ttl": "1s",
+				},
+			},
+		}
+
+		for _, c := range cases {
+			t.Run(c.title, func(t *testing.T) {
+				name := mustCreateUpdateJWTType(t, c.body)
+
+				resp := getJWTType(t, name)
+				assertResponseCode(t, resp, 200)
+
+				assertJwtType(t, resp, c.body)
+			})
+		}
+	})
+
+	t.Run("creating failed", func(t *testing.T) {
+		cases := []struct {
+			title string
+			body  map[string]interface{}
+		}{
+			{
+				title: "without ttl",
+				body: map[string]interface{}{
+					"options_schema": testJwtTypeOptionSchemaValid,
+				},
+			},
+
+			{
+				title: "with ttl less than 1 second",
+				body: map[string]interface{}{
+					"ttl": "0s",
+				},
+			},
+
+			{
+				title: "if options_schema not is openapi3",
+				body: map[string]interface{}{
+					"ttl":            "1s",
+					"options_schema": "invalid",
+				},
+			},
+		}
+
+		for _, c := range cases {
+			t.Run(fmt.Sprintf("returns 400 %s", c.title), func(t *testing.T) {
+				_, resp := createJWTType(t, c.body)
+				assertResponseCode(t, resp, 400)
+			})
+
+			t.Run(fmt.Sprintf("does not creating %s", c.title), func(t *testing.T) {
+				name, _ := createJWTType(t, c.body)
+				resp := getJWTType(t, name)
+				assertResponseCode(t, resp, 404)
+			})
+		}
+	})
+}
+
+func jWTTypeGet(t *testing.T) {
```

Comment:
Can we pick one variation of writing jwt? Now we have Jwt, jWT, JWT.

Reply:
jwt. sory